lib/qesapdeployment.pm

```diff
@@ -867,24 +882,30 @@ sub qesap_delete_az_peering {
 
 =head3 qesap_get_peering_name
 
-    Returns the peering name for the current job, or
+    Search for all network peering related to both:
+     - resource group related to the current job
+     - the provided resource group.
+    Returns the peering name or
     empty string if a peering doesn't exist
 
+=over 1
+
+=item B<RESOURCE_GROUP> - resource group connected to the peering
+
+=back
 =cut
 
 sub qesap_get_peering_name {
-    my $job_id = get_current_job_id();
-    my $json_response = script_output("az network vnet peering list -g dsdm0we --vnet-name vnet-dsdm0we");
-    my $peerings = decode_json($json_response);
-
-    for my $peering (@$peerings) {
-        my $name = $peering->{name};
+    my (%args) = @_;
 
-        if ($name =~ /$job_id/) {
-            return $name;
-        }
-    }
-    return "";
+    my $job_id = get_current_job_id();
+    my $cmd = join(' ', 'az network vnet peering list',
+        '-g', $args{resource_group},
+        '--vnet-name', qesap_get_vnet($args{resource_group}),
```

Comment:
What happens if `$args{resource_group}` is empty or undefined? I don't see a validation for the resource group variable within `qesap_get_vnet()` code either.

Reply:
True. I had it hardcoded, but MIchele's right - it should be variable since the IBSM RG Name could change. But the code retrieves the IBSM resource group using `get_required_var`, so theoretically it should have failed before this code if it was missing.

But I agree with Alvaro, there's no reason to not check for valid arguments here too.